homeassistant/components/loqed/lock.py

```diff
@@ -0,0 +1,269 @@
+        return LOCK_STATES[self.bolt_state] == STATE_JAMMED
+
+    @property
+    def is_locked(self):
+        """Return true if lock is locked."""
+        return LOCK_STATES[self.bolt_state] == STATE_LOCKED
+
+    @property
+    def battery(self):
+        """Return true if lock is locked."""
+        return self._lock.battery_percentage
+
+    @property
+    def extra_state_attributes(self):
+        """Extra state attribtues."""
+        state_attr = {
+            "id": self._lock.id,
+            "bolt_state": self.bolt_state,
+            "lock_url": self.lock_url,
+            "webhook_url": self._webhook,
+            ATTR_BATTERY_LEVEL: self._lock.battery_percentage,
+            "battery_type": self._lock.battery_type,
+            "battery_voltage": self._lock.battery_voltage,
+            "wifi_strength": self._lock.wifi_strength,
+            "ble_strength": self._lock.ble_strength,
+            "last_event": self._lock.last_event,
+            # "party_mode": self._lock.party_mode,
+            # "guest_access_mode": self._lock.guest_access_mode,
+            # "twist_assist": self._lock.twist_assist,
+            # "touch_to_connect": self._lock.touch_to_connect,
+            # "lock_direction": self._lock.lock_direction,
+            # "mortise_lock_type": self._lock.mortise_lock_type,
+            "last_changed_key_id": self._lock.last_key_id,
+        }
+        return state_attr
+
+    async def async_lock(self, **kwargs):
+        """To calls the lock method of the loqed lock."""
+        _LOGGER.debug("start lock operation")
+        await self.async_schedule_update(10)
+        await self._lock.lock()
+
+    async def async_unlock(self, **kwargs):
+        """To call the unlock method of the loqed lock."""
+        _LOGGER.debug("start unlock operation")
+        await self.async_schedule_update(10)
+        await self._lock.unlock()
+
+    async def async_open(self, **kwargs):
+        """To call the open method of the loqed lock."""
+        _LOGGER.debug("start open operation")
+        await self.async_schedule_update(10)
+        await self._lock.open()
+
+    async def async_update(self) -> None:
+        """To update the internal state of the device."""
+        _LOGGER.debug("Start update operation")
+        resp = await self._lock.update()
+        _LOGGER.debug("Update response: %s", str(resp))
+        self._attr_unique_id = self._lock.id
+        self._attr_name = self._lock.name
+        _LOGGER.debug("BOLT_STATE after update: %s", self.bolt_state)
+        self.async_schedule_update_ha_state()
+
+    async def check_webhook(self):
+        """Check if webhook is configured on both sides."""
+        _LOGGER.debug("Start checking webhooks")
+        webhooks = await self._lock.getWebhooks()
+        wh_id = Undefined
+        # Check if hook already registered @loqed
+        for hook in webhooks:
+            if hook["url"].startswith(
+                self._internal_url + "/api/webhook/" + WEBHOOK_PREFIX
```

Comment:
We have helpers to create webhook URLs:

See: `homeassistant.components.webhook` -> `async_generate_url()`